hi-ml-azure/src/health_azure/himl.py

```diff
@@ -831,6 +831,8 @@ def submit_to_azure_if_needed(  # type: ignore
             return _generate_v2_azure_datasets(cleaned_input_datasets, cleaned_output_datasets)
     # This codepath is reached when executing outside AzureML. Here we first check if a script submission to AzureML
     # is necessary. If not, return to the caller for local execution.
+    if submit_to_azureml is None:
+        submit_to_azureml = AZUREML_COMMANDLINE_FLAG in sys.argv[1:]
```

Comment:
There's one more thing we need to do: Before script submission, modify the argument list from `sys.argv` and remove this magic flag (line 926). Otherwise, the argument parser would need to be able to cope with the existence of `--azureml`